tachyon/math/polynomials/multivariate/multilinear_extension_ops.h

```diff
@@ -155,47 +157,76 @@ class MultilinearExtensionOp<MultilinearDenseEvaluations<F, MaxDegree>> {
     return self;
   }
 
-  static MultilinearExtension<D> Div(const MultilinearExtension<D>& self,
-                                     const MultilinearExtension<D>& other) {
+  CONSTEXPR_IF_NOT_OPENMP static std::optional<MultilinearExtension<D>> Div(
+      const MultilinearExtension<D>& self,
+      const MultilinearExtension<D>& other) {
     const std::vector<F>& l_evaluations = self.evaluations_.evaluations_;
     const std::vector<F>& r_evaluations = other.evaluations_.evaluations_;
-    if (r_evaluations.empty()) {
-      // f(x) / 0
-      // TODO(chokobole): It should return std::nullopt.
-      // See https://github.com/kroma-network/tachyon/issues/76.
-      return MultilinearExtension<D>::Zero();
+    // f(x) / 0
+    if (UNLIKELY(r_evaluations.empty() ||
+                 l_evaluations.size() != r_evaluations.size())) {
+      // TODO(ashjeong): implement CUDA error logging
+#if !TACHYON_CUDA
+      LOG(ERROR) << "Division invalid";
+#endif  // TACHYON_CUDA
+      return std::nullopt;
     }
     if (l_evaluations.empty()) {
       // 0 / g(x)
       return self;
     }
-    CHECK_EQ(l_evaluations.size(), r_evaluations.size());
     std::vector<F> o_evaluations(r_evaluations.size());
+    std::atomic<bool> check_valid(true);
     OPENMP_PARALLEL_FOR(size_t i = 0; i < l_evaluations.size(); ++i) {
-      o_evaluations[i] = l_evaluations[i] / r_evaluations[i];
+      std::optional<F> div = l_evaluations[i] / r_evaluations[i];
+      if (UNLIKELY(!div)) {
+        check_valid.store(false, std::memory_order_release);
+        continue;
+      }
+      o_evaluations[i] = std::move(div).value();
+    }
+    if (UNLIKELY(!check_valid.load(std::memory_order_acquire))) {
```

Comment:
See https://stackoverflow.com/a/12448035

```suggestion
    if (UNLIKELY(!check_valid.load(std::memory_order_relax))) {
```